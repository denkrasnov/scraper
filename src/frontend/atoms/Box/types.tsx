import { ReactNode } from "react";

import { Color } from "../Colors/types";

type Display =
  | "block"
  | "inline-block"
  | "inline"
  | "flex"
  | "inline-flex"
  | "none";

type FlexDirection = "column" | "row";

type JustifyContent =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "space-evenly";

export type AlignItems =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch";

type FlexWrap = "wrap" | "nowrap";

type Cursor = "pointer";

export interface BoxProps {
  alignItems?: AlignItems;
  backgroundColor?: Color;
  children?: ReactNode;
  display?: Display;
  flexDirection?: FlexDirection;
  flexWrap?: FlexWrap;
  grow?: number;
  flex?: number;
  justifyContent?: JustifyContent;
  padding?: string;
  margin?: string;
  marginBottom?: string;
  width?: string;
  height?: string;
  minHeight?: string;
  maxWidth?: string;
  cursor?: Cursor;
  marginRight?: string;
  onClick?: () => void;
}
