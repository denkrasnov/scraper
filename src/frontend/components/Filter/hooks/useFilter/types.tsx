export type Filter = { key: string; value: string };

export type UseFilterReturn<T> = [
  T[],
  (checked: boolean, value: string, filterByKey: string) => void
];
