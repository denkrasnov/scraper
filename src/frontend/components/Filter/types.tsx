// TODO: create shared types with /backend
import { ProductName, Article, Channel } from "../../../backend/scrapers/types";
import { Locale } from "../../../types";

export interface FilterProps {
  productName: ProductName;
  options: Channel[];
  items: Article[];
  locale?: Locale;
  onClickLocale?: () => void;
}
