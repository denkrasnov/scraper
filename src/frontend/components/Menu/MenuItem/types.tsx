import { DataAttr } from "~app/common/types";

export interface MenuItemProps {
  src: string;
  text: string;
  onClick: () => void;
  dataAttr?: DataAttr;
}
