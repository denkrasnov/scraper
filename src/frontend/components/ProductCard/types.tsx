export interface ProductCardProps {
  header?: string;
  imageURL?: string;
  newsURL?: string;
  date?: string;
}
