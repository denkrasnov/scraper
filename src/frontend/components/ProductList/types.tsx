import { Product } from "~app/services/fetchProducts/types";

export interface ProductListProps {
  products: Product[];
}
