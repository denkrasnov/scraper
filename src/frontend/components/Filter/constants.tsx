// TODO: created shared types with /backend
import { ProductName } from "../../../backend/scrapers/types";

export const filterName = {
  [ProductName.MD]: "Canal"
};
