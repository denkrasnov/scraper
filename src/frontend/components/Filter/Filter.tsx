import React, {
  FC,
  ChangeEvent,
  Children,
  cloneElement,
  PropsWithChildren
} from "react";

import { Article } from "../../../backend/scrapers/types";
import Box from "~app/atoms/Box";
import Image from "~app/atoms/Image";
import Checkbox from "~app/atoms/Checkbox";
import Card from "~app/atoms/Card";
import Divider from "~app/atoms/Divider";
import Text from "~app/atoms/Text";
import isDesktop from "~app/atoms/hooks/isDesktop";
import { channelLogo } from "~app/common/constants";
import { filterName } from "./constants";
import { FilterProps } from "./types";
import useFilter from "./hooks/useFilter";

const Filter: FC<PropsWithChildren<FilterProps>> = (props) => {
  const { productName, options, children, items, onClickLocale, locale } =
    props;
  const desktop = isDesktop();
  const [filteredItems, filter] = useFilter<Article>(items);

  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
    filter(event.target.checked, event.target.value, "channel");
  };

  const child = Children.only(children);
  const childWithProps = cloneElement(child as any, {
    products: filteredItems
  });

  return (
    <>
      <Box
        display="inlineBlock"
        gridArea="filter"
        marginBottom="s20"
        marginLeft={desktop ? "s20" : undefined}
        marginTop="s20"
      >
        <Card>
          <Box flexDirection="column" padding="s12">
            <Box justifyContent="spaceBetween" marginBottom="s4">
              <Text fontSize="fs14" fontWeight="fw700">
                {filterName[productName]}
              </Text>
              {locale && (
                <Box cursor="pointer" onClick={onClickLocale}>
                  <Text color="TEXT_GRAY" fontSize="fs16" gradientHover>
                    {locale}
                  </Text>
                </Box>
              )}
            </Box>
            <Divider />
            <Box
              flexDirection={desktop ? "column" : undefined}
              justifyContent={desktop ? "flexStart" : "spaceAround"}
            >
              {options.map((option) => (
                <Box key={option} marginTop="s12">
                  <Checkbox
                    label={
                      <Image
                        alt="channel logo"
                        height="30px"
                        src={channelLogo[option]}
                        width="30px"
                      />
                    }
                    name={option}
                    onChange={onChange}
                    styled={option}
                    value={option}
                  />
                </Box>
              ))}
            </Box>
          </Box>
        </Card>
      </Box>
      {childWithProps}
    </>
  );
};

export default Filter;
