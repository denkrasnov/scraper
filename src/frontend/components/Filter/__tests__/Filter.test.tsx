import React from "react";
import { shallow } from "enzyme";

// TODO: created shared types with /backend
import { ProductName } from "../../../../backend/scrapers/types";
import Filter from "..";

describe("Filter", () => {
  it("should render", () => {
    const component = shallow(
      <Filter
        options={["__OPTION_1__", "__OPTION_2__"]}
        productName={ProductName.TV}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
