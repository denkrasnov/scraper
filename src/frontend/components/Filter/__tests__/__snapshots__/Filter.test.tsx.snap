// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render 1`] = `
<Fragment>
  <Box
    display="inlineBlock"
    gridArea="filter"
    marginBottom="s20"
    marginLeft="s20"
    marginTop="s20"
  >
    <Card>
      <Box
        flexDirection="column"
        padding="s12"
      >
        <Box
          marginBottom="s4"
        >
          <Text
            fontSize="fs14"
            fontWeight="fw700"
          >
            Canal
          </Text>
        </Box>
        <Divider />
        <Box
          flexDirection="column"
          justifyContent="flexStart"
        >
          <Box
            key="__OPTION_1__"
            marginTop="s12"
          >
            <Checkbox
              label={
                <Image
                  alt="channel logo"
                  width="30px"
                />
              }
              name="__OPTION_1__"
              onChange={[Function]}
              value="__OPTION_1__"
            />
          </Box>
          <Box
            key="__OPTION_2__"
            marginTop="s12"
          >
            <Checkbox
              label={
                <Image
                  alt="channel logo"
                  width="30px"
                />
              }
              name="__OPTION_2__"
              onChange={[Function]}
              value="__OPTION_2__"
            />
          </Box>
        </Box>
      </Box>
    </Card>
  </Box>
  <div
    products={Array []}
  />
</Fragment>
`;

exports[`Filter should render when mobile 1`] = `
<Fragment>
  <Box
    display="inlineBlock"
    gridArea="filter"
    marginBottom="s20"
    marginTop="s20"
  >
    <Card>
      <Box
        flexDirection="column"
        padding="s12"
      >
        <Box
          marginBottom="s4"
        >
          <Text
            fontSize="fs14"
            fontWeight="fw700"
          >
            Canal
          </Text>
        </Box>
        <Divider />
        <Box
          justifyContent="spaceAround"
        >
          <Box
            key="__OPTION_1__"
            marginTop="s12"
          >
            <Checkbox
              label={
                <Image
                  alt="channel logo"
                  width="30px"
                />
              }
              name="__OPTION_1__"
              onChange={[Function]}
              value="__OPTION_1__"
            />
          </Box>
          <Box
            key="__OPTION_2__"
            marginTop="s12"
          >
            <Checkbox
              label={
                <Image
                  alt="channel logo"
                  width="30px"
                />
              }
              name="__OPTION_2__"
              onChange={[Function]}
              value="__OPTION_2__"
            />
          </Box>
        </Box>
      </Box>
    </Card>
  </Box>
  <div
    products={Array []}
  />
</Fragment>
`;
